onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalInfo.java

```diff
@@ -451,31 +252,28 @@ private String computeTimeLabel(Context context) {
         }
     }
 
-    /**
-     * Computes the arrival status label from the delay (i.e., schedule deviation), where positive
-     * means the bus is running late and negative means the bus is running ahead of schedule
-     *
-     * @param delay schedule deviation, in minutes, for this vehicle where positive
-     *              means the bus is running late and negative means the bus is running ahead of
-     *              schedule
-     * @return the arrival status label based on the deviation
-     */
-    public static String computeArrivalLabelFromDelay(Resources res, long delay) {
-        if (delay > 0) {
-            // Arriving delayed
-            return res.getQuantityString(
-                    R.plurals.stop_info_arrive_delayed, (int) delay,
-                    delay);
-        } else if (delay < 0) {
-            // Arriving early
-            delay = -delay;
-            return res
-                    .getQuantityString(
-                            R.plurals.stop_info_arrive_early,
-                            (int) delay, delay);
+    private String computeNotifyText(Context context) {
+        final String routeName = mInfo.getShortName();
+        if (mEta <= 0) {
+            if (mIsArrival) {
+                return context.getString(R.string.trip_stat_gone_arrived, routeName);
+            } else {
+                return context.getString(R.string.trip_stat_gone_departed, routeName);
+            }
+        } else if (mEta < ONE_MINUTE) {
```

Comment:
Related to the above comment - I think this should be:

`} else if (mEta == 0) {`

With the current values, I don't think this would ever fire - `mEta` is rounded to the minute, and the preceding IF statement would catch any equivalence to 0, so we couldn't have any floating point values between `0` and `1`.

I think with these two changes, it should calculate "NOW" arrivals and departures correctly.

On a related note, I don't think there are currently any unit tests that cover this - we should add some to make sure all the IF branches are covered - we can use another raw mocked response if needed.